File: sample-business/src/main/java/com/teamwork/android/samples/clean/business/injection/module/bridge/BusinessLayerBridgeModule.java
package com.teamwork.android.samples.clean.business.injection.module.bridge;

import com.teamwork.android.samples.clean.business.injection.DaggerSampleBusinessComponent;
import com.teamwork.android.samples.clean.business.injection.SampleBusinessComponent;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

/**
 * Provider Dagger {@link Module} where the {@link SampleBusinessComponent} gets initialized.
 * <p>
 * This module must be be used in the application/presentation layer component to provide the dependencies exposed by
 * the {@link SampleBusinessComponent}.
 */
@Module
public class BusinessLayerBridgeModule {

    @Provides
    @Singleton
    SampleBusinessComponent businessComponent() {
        SampleBusinessComponent dataComponent = DaggerSampleBusinessComponent.builder().build();
        SampleBusinessComponent.getProvider().setComponent(dataComponent);
        return dataComponent;
    }

}

File: sample-business/src/main/java/com/teamwork/android/samples/clean/business/injection/module/bridge/BusinessLayerBridgeModule.kt
package com.teamwork.android.samples.clean.business.injection.module.bridge

import android.content.Context
import com.teamwork.android.samples.clean.business.injection.DaggerSampleBusinessComponent
import com.teamwork.android.samples.clean.business.injection.SampleBusinessComponent
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Provider Dagger [Module] where the [SampleBusinessComponent] gets initialized.
 *
 * This module must be be used in the application/presentation layer component to provide the dependencies exposed by
 * the [SampleBusinessComponent].
 */
@Module
object BusinessLayerBridgeModule {

    @Provides
    @Singleton
    @JvmStatic
    fun businessComponent(applicationContext: Context): SampleBusinessComponent {
        val dataComponent = DaggerSampleBusinessComponent.factory().create(applicationContext)
        SampleBusinessComponent.provider.component = dataComponent
        return dataComponent
    }

}